inveniordm_py/records/resources.py

```diff
@@ -191,26 +196,104 @@ def search(self, q="", page=1, size=10, sort="newest", allversions=False):
         )
 
 
-# TODO implement files
-# class FilesList(Resource):
-#     endpoint = "/records/{id_}/files"
+class RecordFilesList(Resource):
+    """Implements a RecordFilesList as a Resource.
+
+    This is the resource that is used to interact with the /api/records/{id_}/files endpoint.
+    """
+
+    endpoint = "/records/{id_}/files"
+
+    def get(self):
+        """Get a files list."""
+        return self._get(FilesListMetadata)
+
+    def __call__(self, key):
+        """Instantiate a record item resource."""
+        return RecordFile(self._client, filename=key, **self._endpoint_args)
+
+
+class RecordFile(Resource):
+    """Implements a RecordFile as a Resource.
+
+    This is the resource that is used to interact with the /api/records/{id_}/files/{filename} endpoint.
+    """
+
+    endpoint = "/records/{id_}/files/{filename}"
+
+    def get(self):
+        """Get file metadata."""
+        return self._get(FileMetadata)
+
+    def download(self):
+        """Download a file."""
+        return self._get_raw(
```

Comment:
I am not sure about this implementation.

The tool is designed around the concept of a "Resource", which is a 1 to 1 match with RDM's resources (i.e. Records, Drafts, etc). However, to download a file we don't get any RDM-equivalent "resource". That being said, using the resource's HTTP methods wrappers (GET / POST) doesn't cater for this use case.